### PR TITLE
{Network} Desensitize tests for private link `Microsoft.Cache/Redis`

### PR DESCRIPTION
**Pull request by necusjz**
**Related command**

az network private-link-resource
az network private-endpoint-connection

**Description**

Resolved the issue mentioned in:
- https://github.com/Azure/azure-cli-pr/issues/136

Removed previous suppression for credential scan:
- https://github.com/Azure/azure-cli/pull/20867

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py [hunk 1]
```diff
@@ -16,11 +16,29 @@
...
+from azure.cli.testsdk.scenario_tests import AllowLargeResponse, RecordingProcessor
+from azure.cli.testsdk.scenario_tests.utilities import is_text_payload
 
 TEST_DIR = os.path.abspath(os.path.join(os.path.abspath(__file__), '..'))
 
 
+class CredentialReplacer(RecordingProcessor):
+    def process_response(self, response):
+        import json
+        fake_data = "****"
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py:28]: Let's use some more meaningful moniker, like

https://github.com/Azure/azure-cli/blob/25912763773261340f880bab8f5e360ebc5f735c/src/azure-cli-testsdk/azure/cli/testsdk/utilities.py#L157

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py [hunk 2]
```diff
@@ -16,11 +16,29 @@
...
+        sensitive_data = "accessKeys"
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py:29]: `accessKeys` is not data. Better name would be `sensitive_property`.

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py [hunk 3]
```diff
@@ -16,11 +16,29 @@
...
+        if is_text_payload(response) and response["body"]["string"]:
+            try:
+                props = json.loads(response["body"]["string"])
+                if sensitive_data in props["properties"]:
+                    del props["properties"][sensitive_data]
+                    props["properties"][fake_data] = fake_data
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py:35]: This also deletes the key. No need for that.

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py [hunk 4]
```diff
@@ -16,11 +16,29 @@
...
 from azure.cli.command_modules.rdbms.tests.latest.test_rdbms_commands import ServerPreparer
 from azure.cli.command_modules.batch.tests.latest.batch_preparers import BatchAccountPreparer, BatchScenarioMixin
-from azure.cli.testsdk.scenario_tests import AllowLargeResponse
+from azure.cli.testsdk.scenario_tests import AllowLargeResponse, RecordingProcessor
+from azure.cli.testsdk.scenario_tests.utilities import is_text_payload
 
 TEST_DIR = os.path.abspath(os.path.join(os.path.abspath(__file__), '..'))
 
 
+class CredentialReplacer(RecordingProcessor):
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py:25]: It's better to specify which kind of credential it replaces, redis? If it is a general one, put it in `azure.cli.testsdk`.

**Review by jiasli — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

**Pull request merged**